src/UnitTests/Core/Impl/Threading/UIThreadHelper.cs

```diff
@@ -116,6 +117,12 @@ public class UIThreadHelper {
             return args.Exception;
         }
 
+        public Task DoEventsAsync() {
+            return TaskUtilities.IsOnBackgroundThread()
+                ? _application.Dispatcher.InvokeAsync(() => { }, DispatcherPriority.Background).Task
+                : Task.Run(DoEventsAsync);
```

Comment:
If method is called from background thread, it schedules a task in `Dispatcher`. If not, it calls itself from background thread.